src/ui/DocumentPropertiesView.cpp

```diff
@@ -4,52 +4,73 @@
 #include <cstring>
 #include <format>
 
-#include "../commands/ChangeDocumentProperties.h"
+#include "../ui/command/ChangeDocumentProperties.h"
 
 namespace {
 constexpr auto ElementHeight = 20;
 }
 
 void ui::DocumentPropertiesView::update() {
-  auto subRect = Rectangle{rect.x, rect.y, rect.width, ElementHeight};
+  auto subRect = std::make_shared<Rectangle>(rect.x, rect.y, rect.width, ElementHeight);
 
-  auto mouse = GetMousePosition();
+  setFilename(subRect);
+  setDimensions(subRect);
+  setApply(subRect);
+}
+
+void ui::DocumentPropertiesView::setDocumentAndUpdateInfo(
+    std::shared_ptr<Document> newDoc) {
+  doc = std::move(newDoc);
+
+  std::strcpy(filenameText, doc->getFilePath().c_str());
+
+  auto dims = doc->getDimensions();
+
+  auto ws = std::format("{:.0f}", dims.x);
+  std::strcpy(widthText, ws.c_str());
+
+  auto hs = std::format("{:.0f}", dims.y);
+  std::strcpy(heightText, hs.c_str());
+}
+void ui::DocumentPropertiesView::setFilename(std::shared_ptr<Rectangle> subRect) {
```

Comment:
```suggestion
void ui::DocumentPropertiesView::setFilename(Rectangle& subRect) {
```

Podobną zmianę należy zrobić dla pozostałych i usunąć alokowanie Rectangle jako shared_ptr (można myśleć o referencjach w C++ w tym przypadku, podobnie jak o borrowach w rust - pożyczasz na chwilę wartość)